gopls/internal/lsp/template/implementations.go

```diff
@@ -75,25 +77,54 @@ func Diagnose(f file.Handle) []*source.Diagnostic {
 // for definitions, type definitions, and implementations.
 // Results only for variables and templates.
 func Definition(snapshot *cache.Snapshot, fh file.Handle, loc protocol.Position) ([]protocol.Location, error) {
-	x, _, err := symAtPosition(fh, loc)
+	sym, p, err := symAtPosition(fh, loc)
 	if err != nil {
 		return nil, err
 	}
-	sym := x.name
 	ans := []protocol.Location{}
-	// PJW: this is probably a pattern to abstract
-	a := New(snapshot.Templates())
-	for k, p := range a.files {
-		for _, s := range p.symbols {
-			if !s.vardef || s.name != sym {
-				continue
+	switch sym.kind {
+	case protocol.Method:
+		workspaceSymbol, err := workspaceSymbolForMethod(context.Background(), sym, p, snapshot)
+		if err != nil {
+			return nil, err
+		}
+		if workspaceSymbol == nil {
+			return ans, nil
+		}
+		ans = append(ans, workspaceSymbol.Location)
+	default:
+		// PJW: this is probably a pattern to abstract
+		a := New(snapshot.Templates())
+		for k, p := range a.files {
+			for _, s := range p.symbols {
+				if !s.vardef || s.name != sym.name {
+					continue
+				}
+				ans = append(ans, protocol.Location{URI: k, Range: p.Range(s.start, s.length)})
 			}
-			ans = append(ans, protocol.Location{URI: k, Range: p.Range(s.start, s.length)})
 		}
 	}
 	return ans, nil
 }
 
+func workspaceSymbolForMethod(ctx context.Context, sym *symbol, p *Parsed, snapshot source.Snapshot) (*protocol.SymbolInformation, error) {
+	if p.goTypePackage == "" || p.goTypeName == "" {
+		return nil, nil
+	}
+	querySuffix := sym.name
+	if sym.parentNames != nil && len(sym.parentNames) > 0 {
+		querySuffix = strings.Join(sym.parentNames, ".") + "." + querySuffix
+	}
+	query := p.goTypePackage + "." + p.goTypeName + "." + querySuffix
+	// TODO(mortenson) This doesn't actually deeply match fields, which was kind of the hope with storing parentNames.
+	// Maybe we can call WorkspaceSymbols on `p.goTypePackage + "." + p.goTypeName``, then walk from there for each sym.parentNames?
```

Comment:
This is where I'd like help. Couldn't really figure it out from reading `internal/lsp/source/implementation.go`